Update developer's guide and uml diagram (#126)
AY2021S1-CS2113-T13-1/tp: docs/UserGuide.md

Expected output:

> ### <a id="tag-n"><ins>3.13 Tag/Untag Notes:</ins> `tag`</a>

Reviewer: ```suggestion
### <a id="tag-n"><ins>3.13 Tag/Untag Notes:</ins> `tag-n`</a>
```

Tags or untags a note with the given tag name.

Format: `tag-n /i INDEX /tag TAG [TAG COLOR] [/tag TAG]...`

- Tags a note with a given TAG. If the note already contains the TAG, the tag will be removed from the note.The index refers to the index number shown in the displayed note list. The index must be a **positive integer** (1, 2, 3, …).
- The user can create tag or untag multiple tags to a note within a single line.

> 💡 If the user tries to tag a non existing tag to a note, the application will automatically create the tag for it.

Reviewer: ```suggestion
💡 If the user tries to tag a non-existing tag to a note, the application will automatically create a new tag, and add the tag to the note.
```

Example of usage: 

> `Tag-n /i 1 /tag CS2113`
> 
> `Tag-n /i 1 /tag CS2113 /tag important`

Reviewer: ```suggestion
`tag-n /i 1 /tag CS2113`

`tag-n /i 1 /tag CS2113 /tag important`
```

Expected output:

> ### <a id="tag-e"><ins>3.14 Tag/Untag Events:</ins> `tag`</a>

Reviewer: ```suggestion
### <a id="tag-e"><ins>3.14 Tag/Untag Events:</ins> `tag-e`</a>
```

Tags or untags an event with the given tag name.

Format: `tag-e /i INDEX /tag TAG [TAG COLOR] [/tag TAG]...`

- Tags a note with a given TAG. If the event already contains the TAG, the tag will be removed from the event.The index
 refers to the index number shown in the displayed event list. The index must be a **positive integer** (1, 2, 3, …).
- The user can create tag or untag multiple tags to an event within a single line.

> 💡 If the user tries to tag a non existing tag to an event, the application will automatically create the tag for it.

Reviewer: ```suggestion
💡 If the user tries to tag a non-existing tag to a note, the application will automatically create a new tag, and add the tag to the note.
```

Author: This is for tag-e

Reviewer: My apologies, I copy-pasted from the change I made for tag-n then forgot to update it.

Author: Noted. Will edit it

Example of usage: 

> `Tag-e /i 1 /tag CS2113`
> 
> `Tag-e /i 1 /tag CS2113 /tag important`

Reviewer: ```suggestion
`tag-e /i 1 /tag CS2113`

`tag-e /i 1 /tag CS2113 /tag important`
```

Expected output:

### <a id="delete-t"><ins>3.15 Delete Tags:</ins> `delete-t`</a>

> Deletes a tag from the list of tags and remove the tag from the related notes.

Reviewer: Only tags? Can we do it for events as well?

Author: This is deleting of the tag and not untagging. Will update the content.
